ITKImageProcessing/src/ITKImageProcessing/Filters/ITKImportImageStack.cpp

```diff
@@ -241,43 +356,76 @@ Result<> ITKImportImageStack::executeImpl(DataStructure& dataStructure, const Ar
+    readResult = cxITKImportImageStack::ReadImageStack<uint16>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::int16: {
-    return cxITKImportImageStack::ReadImageStack<int16>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<int16>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::uint32: {
-    return cxITKImportImageStack::ReadImageStack<uint32>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<uint32>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::int32: {
-    return cxITKImportImageStack::ReadImageStack<int32>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<int32>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::uint64: {
-    return cxITKImportImageStack::ReadImageStack<uint64>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<uint64>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::int64: {
-    return cxITKImportImageStack::ReadImageStack<int64>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<int64>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::float32: {
-    return cxITKImportImageStack::ReadImageStack<float32>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<float32>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   case NumericType::float64: {
-    return cxITKImportImageStack::ReadImageStack<float64>(dataStructure, imageGeomPath, imageDataPath, files, messageHandler, shouldCancel);
+    readResult = cxITKImportImageStack::ReadImageStack<float64>(dataStructure, imageGeomPath, imageDataPath, files, imageTransformValue, messageHandler, shouldCancel);
+    break;
   }
   default: {
     throw std::runtime_error("Unsupported array type");
   }
   }
+  if(readResult.invalid())
+  {
+    return readResult;
+  }
+
+  //  if(imageTransformValue != k_NoImageTransform)
```

Comment:
Is this going to be used down the line? if not I would remove it

Reply:
It should be used down the line. Not sure why it isn't